### PR TITLE
[DOCUMENTATION] Make it clearer that MessageSizeEstimator is critical to get accurate backpressure

### PR DESCRIPTION
**Pull request by chibenwa**
Motivation:

Without such warning it might be tricky to get backpressure right.

Modification:

Simple documentation

Result:

Simple documentation

### CODE REVIEW

File: transport/src/main/java/io/netty/channel/Channel.java [hunk 1]
```diff
@@ -157,18 +157,25 @@ public interface Channel extends AttributeMap, ChannelOutboundInvoker, Comparabl
      * requested write operation immediately.  Any write requests made when
      * this method returns {@code false} are queued until the I/O thread is
      * ready to process the queued write requests.
+     *
+     * {@link WriteBufferWaterMark} can be used to configure on which coundition
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/Channel.java:161]:
```suggestion
     * {@link WriteBufferWaterMark} can be used to configure on which coundition
```

File: transport/src/main/java/io/netty/channel/WriteBufferWaterMark.java [hunk 2]
```diff
@@ -29,6 +29,9 @@
  * dropped down below the {@linkplain #low low water mark},
  * {@link Channel#isWritable()} will start to return
  * {@code true} again.
+ * <p>
+ * Note that messages needs to be handled by the {@link MessageSizeEstimator}
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/WriteBufferWaterMark.java:33]:
```suggestion
 * Note that messages needs to be handled by the {@link MessageSizeEstimator}
```

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: transport/src/main/java/io/netty/channel/Channel.java [hunk 3]
```diff
@@ -157,18 +157,25 @@ public interface Channel extends AttributeMap, ChannelOutboundInvoker, Comparabl
...
```
Comment (thc202) [inline transport/src/main/java/io/netty/channel/Channel.java:161]:
```suggestion
     * {@link WriteBufferWaterMark} can be used to configure on which condition
```

**Review by normanmaurer — APPROVED** [review]
Thanks a lot!

**Pull request merged**